wicket-core/src/main/java/org/apache/wicket/util/tester/BaseWicketTester.java

```diff
@@ -495,12 +494,21 @@ public void clearFeedbackMessages()
 	 */
 	protected void cleanupFeedbackMessages(IFeedbackMessageFilter filter)
 	{
-		ApplicationSettings applicationSettings = application.getApplicationSettings();
-		IFeedbackMessageFilter old = applicationSettings.getFeedbackMessageCleanupFilter();
-		applicationSettings.setFeedbackMessageCleanupFilter(filter);
-		getLastRenderedPage().detach();
-		getSession().detach();
-		applicationSettings.setFeedbackMessageCleanupFilter(old);
+		
```

Comment:
We cannot detach the session, otherwise it will block access to its page manager afterwards. Actually detachment isn't needed anyways, just the message have to be cleared. For symmetry don't call detach on components either.